task3.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+let primes = [];
+
+function seiveOriginal(n) {
+  let nPrime = new Array(n + 1).fill(false);
+  nPrime[0] = true;
+  nPrime[1] = true;
+  for (let i = 2; i <= n; i++) {
+    if (!nPrime[i]) {
+      primes.push(i);
+      for (let j = i * i; j <= n; j += i) {
+        nPrime[j] = true;
+      }
+    }
+  }
+}
+
+function getPower(divider, i, power, n) {
```

Comment:
This function doesn't really need to know about the `power` array and the index. Just make the function to receive `divider` and `n` and return the power for the specified divider.